pkg/R/RDD.R

```diff
@@ -1294,6 +1294,79 @@ setMethod("sortBy",
             values(sortByKey(keyBy(rdd, func), ascending, numPartitions))
           })
 
+# Helper function to get first N elements from an RDD in the specified order.
+# Param:
+#   rdd An RDD.
+#   num Number of elements to return.
+#   ascending A flag to indicate whether the sorting is ascending or descending.
+# Return:
+#   A list of the first N elements from the RDD in the specified order.
+#
+takeOrderedElem <- function(rdd, num, ascending = TRUE) {          
+  if (num <= 0L) {
+    return(list())
+  }
+  
+  partitionFunc <- function(part) {
+    if (num < length(part)) {
+      # R limitation: order works only on primitive types!
+      ord <- order(unlist(part, recursive = FALSE), decreasing = !ascending)
+      part[ord[1:num]]
+    } else {
+      part
+    }
+  }
+  
+  newRdd <- mapPartitions(rdd, partitionFunc)
+  take(sortBy(newRdd, function(x) { x }, ascending = ascending), num)
```

Comment:
Yeah, I did not realize that scala uses Bounded priority queue. OK I agree I will follow it. Actually pyspark also does so.